jit.js

```diff
@@ -198,13 +380,31 @@ to single-step.
             "    inst = rcvr.pointers,\n",
             "    temp = vm.homeContext.pointers,\n",
             "    lit = vm.method.pointers,\n",
-            "    bytecodes = 0 - vm.pc;\n",
-            "while (true) switch (vm.pc) {\n"
-        );
+            "    bytecodes = 0 - vm.pc;\n");
+
+        this.labelBootstrapStart = this.source.length;
+        this.source.push("while (true) switch (vm.pc) {\n");
+        
         this.done = false;
         while (!this.done) {
-            var byte = method.bytes[this.pc++],
-                byte2 = 0;
+            var byte = method.bytes[this.pc++];
+
+            var peepholes = this.peepholes;
+            if (peepholes.length > 0) {
+                peepholes = peepholes.filter(function (peephole) {
+                    var i;
+                    for (i = 0; i < peephole.byteCount; i += 1) {
+                        if (!peephole.matches(i, method.bytes[this.pc - 1 + i])) {
+                            return false;
+                        }
+                    }
+                    return true;
+                }, this);
+                if (peepholes.length > 0) {
+                    peepholes[0].generate.call(this, method.bytes.subarray(this.pc - 1, this.pc - 1 + peepholes[0].byteCount));
+                }
+            }
+
```

Comment:
This does not work correctly. It generates the peep-holed coded twice. E.g., this return:
```
function Set_add_(vm) {
// Set>>add:
var context = vm.activeContext,
    stack = context.pointers,
    rcvr = vm.receiver,
    inst = rcvr.pointers,
    temp = vm.homeContext.pointers,
    lit = vm.method.pointers,
    bytecodes = 0 - vm.pc;
while (true) switch (vm.pc) {
case 0:
...
case 23:
vm.pc = 25; vm.doReturn(temp[6]); return bytecodes + 25;

stack[++vm.sp] = temp[6];
vm.pc = 25; vm.doReturn(stack[vm.sp]); return bytecodes + 25;
default: bytecodes += vm.pc; vm.interpretOne(true); return bytecodes;}
}
```